File: 2554-maximum-number-of-integers-to-choose-from-a-range-i/2554-maximum-number-of-integers-to-choose-from-a-range-i.java
class Solution {
    public int maxCount(int[] banned, int n, int maxSum) {
        HashSet<Integer> set=new HashSet<>();
        //int len=banned.length;
        int count=0;
        for(int i=0;i<banned.length;i++)
        {
            set.add(banned[i]);
        }
        
        int low=1;
        int sum=0;
        int max=maxSum;
        
        while(low<=n)
        {
            if(set.contains(low))
            {
                low++;
                continue;
            }
            else if(sum+low<=max)
            {
                
                sum+=low;
                count++;
            }
            low++;
        }
        return count;
    }
}

File: 2554-maximum-number-of-integers-to-choose-from-a-range-i/2554-maximum-number-of-integers-to-choose-from-a-range-i.cpp
class Solution {
public:
    int maxCount(vector<int>& banned, int n, int maxSum) {
        unordered_map<int, int> map;
        for(int i=0; i<banned.size();i++)
        {
            map[banned[i]]++;
        }
        
        int count=0;
        int sum=0;
        
        for(int i=1; i<=n;i++)
        {
            if(map.find(i)!=map.end())
            {
                continue;
            }
            else
            {
                if(sum+i<=maxSum)
                {
                    sum+=i;
                    count++;
                }
            }
        }
        return count;
    }
};